plugins/obs-ffmpeg/ffmpeg-encoded-output.c

```diff
@@ -0,0 +1,915 @@
+    Copyright (C) 2014 by Hugh Bailey <obs.jim@gmail.com>
+
+    This program is free software: you can redistribute it and/or modify
+    it under the terms of the GNU General Public License as published by
+    the Free Software Foundation, either version 2 of the License, or
+    (at your option) any later version.
+
+    This program is distributed in the hope that it will be useful,
+    but WITHOUT ANY WARRANTY; without even the implied warranty of
+    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+    GNU General Public License for more details.
+
+    You should have received a copy of the GNU General Public License
+    along with this program.  If not, see <http://www.gnu.org/licenses/>.
+******************************************************************************/
+
+#include "ffmpeg-encoded-output.h"
+
+static const char *ffmpeg_encoded_output_getname(void *unused);
+static void ffmpeg_encoded_output_destroy(void *data);
+static void *ffmpeg_encoded_output_create(obs_data_t * settings,
+					  obs_output_t * output);
+static void ffmpeg_encoded_output_stop(void *data, uint64_t ts);
+static bool ffmpeg_encoded_output_start(void *data);
+static void ffmpeg_encoded_output_data(void *data,
+				       struct encoder_packet *packet);
+static void ffmpeg_encoded_output_defaults(obs_data_t * defaults);
+static obs_properties_t *ffmpeg_encoded_output_properties(void *unused);
+static uint64_t ffmpeg_encoded_output_total_bytes_sent(void *data);
+static int ffmpeg_encoded_output_dropped_frames(void *data);
+static float ffmpeg_encoded_output_congestion(void *data);
+static int ffmpeg_encoded_output_connect_time(void *data);
+
+struct obs_output_info ffmpeg_encoded_output_info = {
+	.id = "ffmpeg_encoded_output",
+	.flags = OBS_OUTPUT_AV |
+	    OBS_OUTPUT_ENCODED | OBS_OUTPUT_SERVICE | OBS_OUTPUT_MULTI_TRACK,
+	.encoded_video_codecs = "h264",
+	.encoded_audio_codecs = "aac",
+	.get_name = ffmpeg_encoded_output_getname,
+	.create = ffmpeg_encoded_output_create,
+	.destroy = ffmpeg_encoded_output_destroy,
+	.start = ffmpeg_encoded_output_start,
+	.stop = ffmpeg_encoded_output_stop,
+	.encoded_packet = ffmpeg_encoded_output_data,
+	.get_defaults = ffmpeg_encoded_output_defaults,
+	.get_properties = ffmpeg_encoded_output_properties,
+	.get_total_bytes = ffmpeg_encoded_output_total_bytes_sent,
+	.get_congestion = ffmpeg_encoded_output_congestion,
+	.get_connect_time_ms = ffmpeg_encoded_output_connect_time,
+	.get_dropped_frames = ffmpeg_encoded_output_dropped_frames
+};
+
+static int proto_init(struct ffmpeg_encoded_output *stream);
+static int proto_try_connect(struct ffmpeg_encoded_output *stream);
+static int proto_connect_time(struct ffmpeg_encoded_output *stream);
+static void proto_close(struct ffmpeg_encoded_output *stream);
+static void proto_set_output_error(struct ffmpeg_encoded_output *stream);
+static int proto_send_packet(struct ffmpeg_encoded_output *stream,
+			     struct encoder_packet *packet, bool is_header);
```

Comment:
Style: No need to align like this. Use two tabs to indent a line continuation.

```suggestion
		struct encoder_packet *packet, bool is_header);
```